File: cannonlaboratory/src/main/java/com/skx/tomike/cannonlaboratory/ui/activity/OutterStartActivity.java
package com.skx.tomike.cannonlaboratory.ui.activity;

import android.content.Intent;
import android.net.Uri;
import android.view.View;

import com.skx.tomike.cannonlaboratory.R;
import com.skx.tomikecommonlibrary.base.BaseViewModel;
import com.skx.tomikecommonlibrary.base.SkxBaseActivity;
import com.skx.tomikecommonlibrary.base.TitleConfig;


public class OutterStartActivity extends SkxBaseActivity<BaseViewModel> {

    @Override
    protected void initParams() {

    }

    @Override
    protected TitleConfig configHeaderTitle() {
        return new TitleConfig.Builder().setTitleText("打开外部APP").create();
    }

    @Override
    protected int getLayoutId() {
        return R.layout.activity_outter_start;
    }

    @Override
    protected void initView() {

    }

    public void startXiaoZhu(View view) {
        Intent intent = new Intent();
        intent.setData(Uri.parse("openapp.xzdz://xiaozhu/openpage?page=index"));
        startActivity(intent);
    }

}

File: cannon/src/main/java/com/skx/tomike/cannon/ui/activity/OutterStartActivity.kt
package com.skx.tomike.cannon.ui.activity

import android.content.Intent
import android.net.Uri
import android.widget.TextView
import com.alibaba.android.arouter.facade.annotation.Route
import com.skx.common.base.BaseViewModel
import com.skx.common.base.SkxBaseActivity
import com.skx.common.base.TitleConfig
import com.skx.common.utils.ToastTool
import com.skx.tomike.cannon.R
import com.skx.tomike.cannon.ROUTE_PATH_OUTER_START

@Route(path = ROUTE_PATH_OUTER_START)
class OutterStartActivity : SkxBaseActivity<BaseViewModel>() {

    override fun initParams() {}

    override fun configHeaderTitle(): TitleConfig {
        return TitleConfig.Builder().setTitleText("打开外部APP").create()
    }

    override fun layoutId(): Int {
        return R.layout.activity_outter_start
    }

    override fun initView() {
        findViewById<TextView>(R.id.tv_outterStart_to).setOnClickListener {
            try {
                val intent = Intent()
                intent.data = Uri.parse("openapp.xzdz://xiaozhu/openpage?page=index")
                startActivity(intent)
            } catch (e: Exception) {
                ToastTool.showToast(this, e.message)
                e.printStackTrace()
            }
        }
    }
}